src/importer.test.ts

```diff
@@ -281,3 +283,91 @@ test('manually importing files', async () => {
   expect(fs.existsSync(filenames.original)).toBe(true)
   expect(fs.existsSync(filenames.normalized)).toBe(true)
 })
+
+test('scanning pauses on adjudication then continues', async () => {
+  const scanner: jest.Mocked<Scanner> = {
+    scanSheets: jest.fn(),
+  }
+  const store = await Store.memoryStore()
+
+  const importer = new SystemImporter({
+    ...importDirs.paths,
+    store,
+    scanner,
+  })
+
+  await importer.configure(election)
+
+  jest.spyOn(store, 'deleteSheet')
+
+  jest
+    .spyOn(store, 'addSheet')
+    .mockImplementationOnce(async () => {
+      return 'sheet-1'
+    })
+    .mockImplementationOnce(async () => {
+      jest.spyOn(store, 'adjudicationStatus').mockImplementation(async () => {
+        return { adjudicated: 0, remaining: 1 }
+      })
+      return 'sheet-2'
+    })
+    .mockImplementationOnce(async () => {
+      return 'sheet-3'
+    })
+
+  scanner.scanSheets.mockImplementationOnce(async function* (): AsyncGenerator<
+    SheetOf<string>
+  > {
+    yield [
+      join(sampleBallotImagesPath, 'sample-batch-1-ballot-1.jpg'),
+      join(sampleBallotImagesPath, 'blank-page.png'),
+    ]
+
+    yield [
+      join(sampleBallotImagesPath, 'sample-batch-1-ballot-2.jpg'),
+      join(sampleBallotImagesPath, 'blank-page.png'),
+    ]
+
+    yield [
+      join(sampleBallotImagesPath, 'sample-batch-1-ballot-3.jpg'),
+      join(sampleBallotImagesPath, 'blank-page.png'),
+    ]
+  })
+
+  await importer.startImport()
+  await importer.waitForEndOfBatchOrScanningPause()
+
+  expect(store.addSheet).toHaveBeenCalledTimes(2)
+
+  // even if we wait a couple more seconds, no change, we're paused
+  await sleep(2000)
```

Comment:
Do we need to actually spend this time waiting? Could it be 1ms?

Reply:
good call, tested this with sleep(1) and it caught the synthetic bug I introduced.